### PR TITLE
Download latest release from github

### PR DESCRIPTION
**Pull request by fracsi**
Get latest version name from github.

### CODE REVIEW

File: src/TailwindBinary.php [hunk 1]
```diff
@@ -89,6 +89,17 @@ private function downloadExecutable(): void
...
     }
 
+    private function getLatestVersion(): string
+    {
+        try {
+            $response = $this->httpClient->request('GET', 'https://api.github.com/repos/tailwindlabs/tailwindcss/releases/latest');
+
+            return $response->toArray()['name'] ?? self::VERSION;
+        } catch (\Throwable) {
+            return self::VERSION;
```
Comment (bocharsky-bw) [inline src/TailwindBinary.php:99]: How about rename this constant to `DEFAULT_VERSION` (or `FALLBACK_VERSION`) for clarity?

Though, we want to allow users to specify their own version via config, and probably this fallback option would be even redundant then IMO. Thoughts? :)

**Review by bocharsky-bw — COMMENTED** [review]
This is cool! Just a little comment.

It might be a good idea to allow people to choose the version they want to use in their projects, i.e. for some reason you may want to use a legacy version, so it would be awesome to allow it from the config, and we even have an issue related to it: https://github.com/SymfonyCasts/sass-bundle/issues/3

Though, that's an edge case and I'm fine to merge this PR and do that in a different one. Do you want to work on this feature too?

**Comment by fracsi** [comment]
@bocharsky-bw I think if you need a different version, there is the option to use a [custom binary](https://symfony.com/bundles/TailwindBundle/current/index.html#using-a-different-binary).

My original thought about this PR was to eliminate the need for a PR every time tailwind releases a new version - see #16 or df70d0d1dd15a61636a479722e25fef2a4c8fe87

File: src/TailwindBinary.php (hunk 1 above)
Comment (fracsi) [reply src/TailwindBinary.php:99]: I was too hesitating how to name it. I decided to stick with the original name, hoping that someone can come up with a better alternative.

Comment (fracsi) [reply src/TailwindBinary.php:99]: @bocharsky-bw `DEFAULT_VERSION` sounds more acceptable for me. WDYT?

Comment (bocharsky-bw) [reply src/TailwindBinary.php:99]: Yeah, agree on `DEFAULT_VERSION`... at least let's start with it, it's a private const anyway :)

**Comment by bocharsky-bw** [comment]
hm, custom binary may be a good workaround if you need to rollback, though a bit more work on it than just specifying a version you want in the config. I wonder what @weaverryan thinks about it and if that https://github.com/SymfonyCasts/sass-bundle/issues/3 could be closed with this PR.

**Comment by fracsi** [comment]
@bocharsky-bw @weaverryan 
How about the following:
There can be a config option for version (`null` by default)
If `null` - we try to download the latest version
If configured - we try to download the given version

This way there will be no constant default version and fallback.

I have created a seperate PR for this: #37

File: src/TailwindBinary.php [hunk 2]
```diff
@@ -89,6 +89,19 @@ private function downloadExecutable(): void
         chmod($targetPath, 0777);
     }
 
+    private function getLatestVersion(): string
+    {
+        try {
+            $response = $this->httpClient->request('GET', 'https://api.github.com/repos/tailwindlabs/tailwindcss/releases/latest');
+
+            return $response->toArray()['name'] ?? self::DEFAULT_VERSION
+    ;
```
Comment (bocharsky-bw) [inline src/TailwindBinary.php:98]: Let's move it to the previous line to fix CS here :)

File: src/TailwindBinary.php [hunk 3]
```diff
@@ -89,6 +89,19 @@ private function downloadExecutable(): void
...
+        } catch (\Throwable) {
+            return self::DEFAULT_VERSION
```
Comment (bocharsky-bw) [inline src/TailwindBinary.php:100]: Same :)

**Review by bocharsky-bw — APPROVED** [review]
> There can be a config option for version (null by default)
If null - we try to download the latest version
If configured - we try to download the given version

Yes., exactly what we had in mind! And that would be very user-friendly :) Let's do it, I will check your PR a bit later, thanks for create a new PR BTW... but this one is ready to be merged I think, thanks!

**Comment by fracsi** [comment]
> Yes., exactly what we had in mind! And that would be very user-friendly :) Let's do it, I will check your a bit PR later, thanks for create a new PR BTW... but this one is ready to be merged I think, thanks!

If this gets merged, than i can work on the other one to contain only the changes for the version config.

**Comment by bocharsky-bw** [comment]
Hey @fracsi !

I just came up with a possible problem with this solution. Let me explain: when you first install and sue this bundle - it will download the latest Tailwind at the current moment... but later when a new version of Tailwind is released - this bundle will still use the downloaded version (which is not latest anymore). I think, to overcome this, we should include the version number in the binary path, right? something like `/path/to/tailwind/binary/3.3.6/...` - this way when we will have a new Tailwind release - it will be downloaded in the new directory

**Comment by fracsi** [comment]
@bocharsky-bw 
This can work only if we leave the version to default to latest. I can give it a go, if you want.
Maybe creating a version dir can be enough to solve this.

**Comment by bocharsky-bw** [comment]
> Maybe creating a version dir can be enough to solve this

I think so. Yes, feel free to start when you can :)

**Comment by fracsi** [comment]
> I think so. Yes, feel free to start when you can :)

@bocharsky-bw See #38

**Pull request merged**